File: app/src/main/java/com/platdmit/simplecloudmanager/app/adapters/ActionListAdapter.java
package com.platdmit.simplecloudmanager.app.adapters;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import com.platdmit.simplecloudmanager.R;
import com.platdmit.simplecloudmanager.domain.models.Action;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class ActionListAdapter extends RecyclerView.Adapter<ActionListAdapter.ActionListHolder> {

    private List<Action> elementList;

    @NonNull
    @Override
    public ActionListHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        int layoutType = R.layout.fragment_actions_item;
        return new ActionListHolder(layoutInflater, parent, layoutType);
    }

    @Override
    public void onBindViewHolder(@NonNull ActionListHolder holder, int position) {
        holder.bindData(elementList.get(position));
    }

    @Override
    public int getItemCount() {
        return elementList.size();
    }

    public void setContentData(List<Action> elements){
        this.elementList = elements;
    }

    class ActionListHolder extends RecyclerView.ViewHolder{

        private TextView mId = itemView.findViewById(R.id.action_id);
        private TextView mStartedAt = itemView.findViewById(R.id.action_start);
        private TextView mCompletedAt = itemView.findViewById(R.id.action_complete);
        private TextView mType = itemView.findViewById(R.id.action_type);
        private TextView mInitiator = itemView.findViewById(R.id.action_initiator);
        private TextView mStatus = itemView.findViewById(R.id.action_status);

        public ActionListHolder(LayoutInflater inflater, ViewGroup parent, int layoutType) {
            super(inflater.inflate(layoutType, parent, false));
        }

        @SuppressLint("SetTextI18n")
        public void bindData(Action data){
            mId.setText(String.valueOf(data.getId()));
            mStartedAt.setText(data.getStartedAt());
            mCompletedAt.setText(" - "+data.getCompletedAt());
            mType.setText(data.getType());
            mInitiator.setText(data.getInitiator());
            mStatus.setText(data.getStatus());
        }
    }
}

File: mod_servers/src/main/java/com/platdmit/mod_servers/screens/server/tabaction/ActionListAdapter.kt
package com.platdmit.mod_servers.screens.server.tabaction

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.platdmit.simplecloudmanager.R
import com.platdmit.mod_servers.screens.server.tabaction.ActionListAdapter.ActionListHolder
import com.platdmit.mod_servers.domain.models.Action

class ActionListAdapter : RecyclerView.Adapter<ActionListHolder>() {
    private val elementList: MutableList<Action> = mutableListOf()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ActionListHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val layoutType = R.layout.fragment_actions_item
        return ActionListHolder(layoutInflater, parent, layoutType)
    }

    override fun onBindViewHolder(holder: ActionListHolder, position: Int) {
        holder.bindData(elementList[position])
    }

    override fun getItemCount(): Int {
        return elementList.size
    }

    fun setContentData(elements: List<Action>) {
        elementList.clear()
        elementList.addAll(elements)
    }

    inner class ActionListHolder(inflater: LayoutInflater, parent: ViewGroup?, layoutType: Int
    ) : RecyclerView.ViewHolder(inflater.inflate(layoutType, parent, false)) {
        private val mId = itemView.findViewById<TextView>(R.id.action_id)
        private val mStartedAt = itemView.findViewById<TextView>(R.id.action_start)
        private val mCompletedAt = itemView.findViewById<TextView>(R.id.action_complete)
        private val mType = itemView.findViewById<TextView>(R.id.action_type)
        private val mInitiator = itemView.findViewById<TextView>(R.id.action_initiator)
        private val mStatus = itemView.findViewById<TextView>(R.id.action_status)

        @SuppressLint("SetTextI18n")
        fun bindData(data: Action) {
            mId.text = data.id.toString()
            mStartedAt.text = data.startedAt
            mCompletedAt.text = " - ${data.completedAt}"
            mType.text = data.type
            mInitiator.text = data.initiator
            mStatus.text = data.status
        }
    }
}